lib/rules.js

```diff
@@ -12,114 +12,103 @@
+     * @param {string} [rulesDir] Path to rules directory, may be relative. Defaults to `lib/rules`.
+     * @param {string} cwd Current working directory
+     * @returns {void}
+     */
+    load(rulesDir, cwd) {
+        const newRules = loadRules(rulesDir, cwd);
 
-            define(qualifiedRuleId, rule);
+        Object.keys(newRules).forEach(ruleId => {
+            this.define(ruleId, newRules[ruleId]);
         });
     }
-}
 
-/**
- * Access rule handler by id (file name).
- * @param {string} ruleId Rule id (file name).
- * @returns {Function} Rule handler.
- */
-function getHandler(ruleId) {
-    if (typeof rules[ruleId] === "string") {
-        return require(rules[ruleId]);
+    /**
+     * Registers all given rules of a plugin.
+     * @param {Object} plugin The plugin object to import.
+     * @param {string} pluginName The name of the plugin without prefix (`eslint-plugin-`).
+     * @returns {void}
+     */
+    importPlugin(plugin, pluginName) {
+        if (plugin.rules) {
+            Object.keys(plugin.rules).forEach(ruleId => {
+                const qualifiedRuleId = `${pluginName}/${ruleId}`,
+                    rule = plugin.rules[ruleId];
+
+                this.define(qualifiedRuleId, rule);
+            });
+        }
     }
-    return rules[ruleId];
 
-}
+    /**
+     * Access rule handler by id (file name).
+     * @param {string} ruleId Rule id (file name).
+     * @returns {Function} Rule handler.
+     */
+    get(ruleId) {
+        if (typeof this._rules[ruleId] === "string") {
+            return require(this._rules[ruleId]);
+        }
+        return this._rules[ruleId];
 
-/**
- * Get an object with all currently loaded rules
- * @returns {Map} All loaded rules
- */
-function getAllLoadedRules() {
-    const allRules = new Map();
+    }
 
-    Object.keys(rules).forEach(name => {
-        const rule = getHandler(name);
+    /**
+     * Get an object with all currently loaded rules
+     * @returns {Map} All loaded rules
+     */
+    getAllLoadedRules() {
+        const allRules = new Map();
 
-        allRules.set(name, rule);
-    });
-    return allRules;
-}
+        Object.keys(this._rules).forEach(name => {
+            const rule = this.get(name);
 
-/**
- * Reset rules storage.
- * Should be used only in tests.
- * @returns {void}
- */
-function testClear() {
-    rules = Object.create(null);
-}
+            allRules.set(name, rule);
+        });
+        return allRules;
+    }
 
-module.exports = {
-    define,
-    load,
-    importPlugin,
-    get: getHandler,
-    getAllLoadedRules,
-    testClear,
+    /**
+     * Reset rules storage.
+     * Should be used only in tests.
+     * @returns {void}
+     */
+    testClear() {
+        this._rules = Object.create(null);
+    }
 
     /**
-     * Resets rules to its starting state. Use for tests only.
+     * Reset rules storage.
+     * Should be used only in tests.
      * @returns {void}
      */
     testReset() {
```

Comment:
Is `testReset` needed anymore? It seems like it would be better to create a new instance instead.